File: programovani/ulohy/files/modelovani-obrazcu-iii/kosoctverec.cpp
#include <iostream>
using namespace std;


int main() {

    // Deklarace a inicializace
    const int vel = 4; // Inicializace velikosti kosoctverce    
    int pom = vel - 1; // Inicializace pomocne promenne

    // Vnejsi cyklus pro radky
    for(int i = 1; i <= vel; i++) {
        
        // Vnitrni cyklus pro jednotlive hvezdicky
        for(int j = 1; j <= vel + pom; j++) {

            // Pro zacatecni prazdne bunky
            if(pom >= j) {

                // Vypiseme pouze mezeru
                cout << "  ";

            }

            // Pokud jsme jiz v limitu
            else {

                // Vypiseme hvezdicku
                cout << "* ";

            }

        }            

        // Vypiseme novou radku
        cout << endl;

        // Pridame sloupec pocitadlu
        pom--;

    }

}

File: programovani/ulohy/files/modelovani-obrazcu-iii/kosoctverec.js
// Deklarace a inicializace
const vel = 4; // Inicializace velikosti kosoctverce    
pom = vel - 1; // Inicializace pomocne promenne

// Vnejsi cyklus pro radky
for(var i = 1; i <= vel; i++) {

    // Inicalizujeme promennou pro vypsani radky
    var rad = "";
    
    // Vnitrni cyklus pro jednotlive hvezdicky
    for(var j = 1; j <= vel + pom; j++) {

        // Pro zacatecni prazdne bunky
        if(pom >= j) {

            // Pridame do textoveho retezce mezeru
            rad += "  ";

        }

        // Pokud jsme jiz v limitu
        else {

            // Pridame do textoveho retezce hvezdicku
            rad += "* ";

        }

    }            

    // Vypiseme radku
    console.log(rad);

    // Pridame sloupec pocitadlu
    pom--;

}
